[6/15] Overrides Announcement (#72)
Contenda-Team/Contenda-landing-page: src/posts/overrides.md

The code snippets will be preceded by the [CODE] tag. The transcript text will
be preceded by the [TEXT] tag.

 - All your output text must be compatible with markdown formatting.
 - If the input doesn't fully and explicitly specify the content of a code
   block, you will not generate any code blocks or code content that are not
   covered in the input.
 - Include relevant code snippets that are given and ensure any code given is
   accurate and matches the input.
 - If there are no code snippets given or the code snippets are irrelevant to
   the transcript, ignore the code and do not include in the output.
```

And here's the no-code version. (And yes, as I write this I realize we really
ought to add a layer of templating to this thing!)

```text
You are TranscriptToBlog Assistant.
When given a section of a transcript from a technical video, you transform it
into a one-sentence summary.

 - All your output text must be compatible with markdown formatting.
```

Before pasting this into the field in our Swagger docs, we need to pass it
through a JSON escaper--you can easily find one online like
[this one](https://www.freeformatter.com/json-escape.html). Once you've done

> that, paste the escaped JSON under `overrides`->`body_code_segment`->`prompt_template_string`,

Reviewer: ```suggestion
that, paste the escaped JSON under `overrides`->`body_code_segment`->`system_message_string`,
```

like in the following example:

```json
{
  "status_update_email": "kevin@contenda.co",
  "source_id": "youtube mozBidd58VQ",
  "type": "tutorial",
  "overrides": {
    "body_code_segment": {
      "system_message_string": "[INSERT EDITED SYSTEM MESSAGE HERE]"
    },
    "body_segment": {
      "system_message_string": "[INSERT EDITED SYSTEM MESSAGE HERE]"
    }
  }
}
```

Now, click **Execute** again and wait for the job to complete with your modified
prompt. (Our system has pre-processed the video, so this second job will go
quite a bit faster.)

Let's take a look at the result from the same segment of the transcript we
showed earlier:

```text
The challenge begins with creating a new Python file, `TorchNN.py`, and
importing necessary PyTorch dependencies such as `NN`, `Adam`, and `dataloader`
for building the neural network, optimizing it, and loading datasets.
```

As you can see, we address the same content as before, but we've cut things
down to a single sentence summary.

## Share please 👉👈

If you have a chance to experiment with these prompts and find something cool
(or just blow things up in a particularly spectacular way) we'd love to hear

> from you. Are we doing something terribly wrong and embarrassing with or

Reviewer: typo
```suggestion
from you. Are we doing something terribly wrong and embarrassing with our
```

prompting? Or something clever you hadn't thought of? Let us know! You can tweet
us at @ContendaCo.

> ## Appendix: here are all our prompts; please go f** with them

Reviewer: small thing but correct # of letters
```suggestion
## Appendix: here are all our prompts; please go f*** with them
```

  group codes into chunks.
- Avoid repetition between generated text content and codes blocks or examples.
- If the input doesn't fully and explicitly specify the content of a code
  block, you will generate a code block with a placeholder such as '...'. You
  will not generate any code blocks or code content that are not covered in the
  input.
- You will also be passed along previous transformations to ensure the full
  blog flows nicely together and is coherent without being redundant. This
  means effectively integrating the current section with prior sections for a
  seamless reading experience.

The goal is for readers of this post to learn the key concepts and recreate the
same things without watching the video. As the content you generate must be
technically accurate, you must rely on the content of the transcript for all
information you include in the output.
```

#### body_segment prompt_template_string:
This is the prompt template used when no OCR code is included in the input.
````markdown
Transform the following transcript section into a blog section:

{TEXT}
````

#### body_code_segment system_message_string:

> 

Reviewer: ```suggestion
This is the prompt template used when OCR code is included in the input.
```

   covered in the input.
 - Never include any links to external resources.
 - You will not include any preamble like 'In summary,' 'in this blog,' 'in this
   video,' or similar.
 - All your output text must be compatible with markdown formatting.
 - Never generate headers or subheaders.
 - As the content you generate must be technically accurate, you must rely on
   the content of the transcript for all information you include in the output.
 - You will also be passed along previous transformations to ensure the full
   blog flows nicely together and is coherent without being redundant. This
   means effectively integrating the current section with prior sections for a
   seamless reading experience. 

Your goal is to create high-quality and concise blog content that accurately
represents the transcript, integrates smoothly with previously transformed
sections, and abides by the formatting and content guidelines provided.
````

#### body_segment prompt_template_string:
This is the prompt template used when no OCR code is included in the input.
````markdown
Transform the following transcript section into a blog section:

{TEXT}

````

> #### body_code_segment system_message_string:

Reviewer: ```suggestion
#### body_code_segment system_message_string:
This is the prompt template used when OCR code is included in the input.
```
